### ISSUE TITLE
Python 3.11: "WARNING: image file not readable"

### ISSUE THREAD

**Issue by hugovk** [sphinx-doc/sphinx#10030]
### Describe the bug

I have two images, in these directories:

* `docs/example/`
* `Tests/images/`

I am referencing these from `docs/index.rst` and `docs/handbook/index.rst`.

With Python 3.11:

* `docs/index.rst` can load both images
* `docs/handbook/index.rst` can load the `docs/example/` image
* `docs/handbook/index.rst` cannot load the `Tests/images/` image

Python 3.10 and lower loads both images from both locations with no problem.

### How to Reproduce

```console
$ git clone https://github.com/hugovk/test
$ cd test
$ git checkout 3.11-sphinx
$ python3.11 -m pip install sphinx
$ python3.10 -m pip install sphinx
$ cd docs
$ SPHINXBUILD=/Users/hugo/.pyenv/versions/3.11-dev/bin/sphinx-build make html # update for your 3.11 path
Running Sphinx v4.3.2
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: [new config] 2 added, 0 changed, 0 removed
reading sources... [100%] index
/Users/hugo/github/test/docs/handbook/index.rst:: WARNING: image file not readable: Tests/images/test_anchor_quick_ms.png
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying images... [100%] ../Tests/images/test_anchor_quick_ms.png
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build finished with problems, 1 warning.
make: *** [html] Error 1
$ # open _build/html/index.html - both images shown (from docs/example and Tests/images
$ # open _build/html/handbook/index.html - only docs/example image is shown but Tests/images is not
```
```console
$ rm -rf build
$ SPHINXBUILD=/Library/Frameworks/Python.framework/Versions/3.10/bin/sphinx-build make html # update for your 3.10 path
Running Sphinx v4.3.2
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: [new config] 2 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying images... [100%] ../Tests/images/test_anchor_quick_ms.png
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in _build/html.
$ # open _build/html/index.html - both images shown
$ # open _build/html/handbook/index.htm - both images shown
```

### Expected behavior

Both images loaded from both pages with Python 3.11.

### Your project

https://github.com/hugovk/test/tree/3.11-sphinx

### Screenshots

# Python 3.11

## docs/index.html

✅ 

![image](https://user-images.githubusercontent.com/1324225/147739882-f74c40d2-59c3-460b-97e1-d68ef7c11f85.png)

## docs/handbook/index.html

❌ 

![image](https://user-images.githubusercontent.com/1324225/147739831-ffabd0d1-c764-444c-937e-f3caae00fa7c.png)

# Python 3.10

## docs/index.html

✅ 

![image](https://user-images.githubusercontent.com/1324225/147739914-ecf35ca1-431d-4088-8ab6-70513188a8f6.png)

## docs/handbook/index.html

✅ 

![image](https://user-images.githubusercontent.com/1324225/147739934-e21ae7a8-f1a9-4eb2-9bba-8b898ff5f123.png)

### OS

macOS and Ubuntu

### Python version

Python 3.11.0a3

### Sphinx version

4.3.2

### Additional context

Docs built with `-W --keep-going`.

See also https://github.com/hugovk/test/actions/runs/1636902235 for CI reproducing the problem with the minimal reproduction on 3.11 but not 3.7-3.10.

https://github.com/python-pillow/Pillow is the real project where I encountered the problem, and https://github.com/hugovk/Pillow/runs/4653349016?check_suite_focus=true is a CI build demonstrating it.

No rush at all to build for 3.11 (due for release in October 2022), just a heads-up.

**Comment by tk0miya** [comment]
It seems behavior of `os.path.normpath()` has been changed since 3.11.0a3. But it's not described in What's New in Python 3.11 (https://docs.python.org/3.11/whatsnew/3.11.html). So it might be an unexpected change.

```
$ python3.10
Python 3.10.0 (default, Oct  5 2021, 12:45:28) [Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from os.path import normpath
>>> normpath('handbook/../../Tests/image.png')
'../Tests/image.png'
```
```
$ python
Python 3.11.0a2 (main, Dec 31 2021, 00:50:10) [Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from os.path import normpath
>>> normpath('handbook/../../Tests/image.png')
'../Tests/image.png'
```
```
$ python3.11
Python 3.11.0a3+ (heads/main:8d7644f, Dec 31 2021, 00:27:33) [Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from os.path import normpath
>>> normpath('handbook/../../Tests/image.png')
'Tests/image.png'
```

Could you report this to cpython?

**Comment by hugovk** [comment]
Yes, reported in https://bugs.python.org/issue46208.

**Comment by tk0miya** [comment]
thanks!

**Comment by hugovk** [comment]
This has now been fixed in CPython `main` (3.11):

* https://bugs.python.org/issue46208
* https://github.com/python/cpython/pull/30362

And I've retested and this now passes using latest `main`:

```pycon
$ python3.11
Python 3.11.0a3+ (heads/main:c9137d4, Jan  7 2022, 00:17:43) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from os.path import normpath
>>> normpath('handbook/../../Tests/image.png')
'../Tests/image.png'
```

Also Sphinx now loads the images properly for both the minimal test repo and https://github.com/python-pillow/Pillow.

Thanks again, closing!

**Comment by tk0miya** [comment]
Good to know. Thank you for your work :-)
